app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+require 'bcrypt'
+
+class User
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :username, String
+  property :name, String
+  property :email, String, required: true
+  property :password_digest, Text
+
+  has n, :peeps
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(username, password)
+    user = User.first(username: username)
+    if user && BCrypt::Password.new(user.password_digest) == password
+      user
+    else
+      nil
+    end
+  end
+
+  validates_confirmation_of :password
+  validates_uniqueness_of :email
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.